src/models/storage.ts

```diff
@@ -3,6 +3,79 @@
 /// <reference path="./interface.ts" />
 /// <reference path="./otp.ts" />
 
+class BrowserStorage {
+  private static getStorageLocation() {
+    if (localStorage.storageLocation !== 'sync' &&
+        localStorage.storageLocation !== 'local') {
+      return new Promise((resolve, reject) => {
+        let sync: number;
+        let local: number;
+        chrome.storage.local.get((l) => {
+          local = Object.keys(l).length;
+          try {
+            chrome.storage.sync.get((s) => {
+              sync = Object.keys(s).length;
+              // If storage location can't be found try to auto-detect storage
+              // location
+              if (local > sync && sync === 0) {
+                localStorage.storageLocation = 'local';
+              } else if (local < sync && local === 0) {
+                localStorage.storageLocation = 'sync';
+              } else {
+                // No or duplicate user data, use defualt
+                if (navigator.userAgent.indexOf('Edge') !== -1) {
+                  localStorage.storageLocation = 'local';
+                } else {
+                  localStorage.storageLocation = 'sync';
+                }
+              }
+              resolve(localStorage.storageLocation);
```

Comment:
It is better always return after resolve or reject in a promise. `return resolve(...)` or `resolve(...); return;`